src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -530,13 +545,22 @@ def update_existing_check_in(
                     monitor_environment=monitor_environment,
                     guid=guid,
                 )
+
+                # Race condition. The check-in was created (such as an
+                # in_progress) while this check-in was being processed.
+                # Create a new one now.
+                #
+                # XXX(epurkhiser): Is this needed since we're already
+                # locking this entire process?
```

Comment:
Possibly not but want to leave it in until we get a handle on multi-consumer processing